### PR TITLE
feature/coupled-crow:   update model, use netcdf atm output, updates for wave post

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
This PR has various updates in preparation for prototype 6 runs including: 
* Updates to use the most recent version of ufs-weather-model which required changes from "v16beta"->"v16" in the CCPP suite files and a temporary build fix for orion. 
* Does not link un-needed files during wave prep for fully coupled (Fixes Issue #236) 
* Adds separate job resources for wave post jobs (Issue #209) 
* Uses netcdf instead of nemsio output for the atm model (Issue #250) 
* Updates a few namelist variables for p6 configuration including adding dz_min, turning on nst_anl and updating FNMSKH

A test on hera has been made, see COM: 
/scratch2/NCEPDEV/climate/Jessica.Meixner/p6workflow/test02/COMROOT/test02/gfs.20130401/00 the atm grib files should be double checked by someone else. 

This PR should close the issues mentioned: #209, #236, #250

### CODE REVIEW

**Comment by JessicaMeixner-NOAA** [comment]
@LydiaStefanova-NOAA could you check the atm grib files here: /scratch2/NCEPDEV/climate/Jessica.Meixner/p6workflow/test02/COMROOT/test02/gfs.20130401/00

**Comment by LydiaStefanova-NOAA** [comment]
@JessicaMeixner-NOAA  The sfluxgrbf files look good. The rest (pgrb2.1p00, pgrb2b.1p00, flux.1p00, pgrb2.0p25 and pgrb2b.0p25) are all empty.

**Comment by JessicaMeixner-NOAA** [comment]
Thanks @LydiaStefanova-NOAA  I converted this PR to a draft so we can find and fix this issue before merging

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by JessicaMeixner-NOAA** [comment]
@LydiaStefanova-NOAA please check: /scratch2/NCEPDEV/climate/Jessica.Meixner/p6netcdf/testnetcdf01/COMROOT/netcdftest02/gfs.20130401/00/atmos I believe all issues with atm post have been resolved.

**Comment by LydiaStefanova-NOAA** [comment]
@JessicaMeixner-NOAA All atm grib2 files are looking good!

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA this is ready for re-review (a fair amount has changed since you gave it the thumbs up) and is ready to be merged when you think it's acceptable.

File: workflow/defaults/settings.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ default_settings: &default_settings
   nco_mode: NO
   use_nco_ecflow_headers: NO
   print_esmf: .false.
+  esmf_profile: .flase.
```
Comment (WalterKolczynski-NOAA) [inline workflow/defaults/settings.yaml]: typo

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
One typo noted.

Also, be more careful with indentation in the future. I know the current codebase already is terrible with inconsistent indentation, I'd like to not make the problem worse after this (bonus points if you fix existing indentation issues in future PRs).

File: workflow/defaults/settings.yaml (hunk 1 above)
Comment (JessicaMeixner-NOAA) [reply workflow/defaults/settings.yaml]: This typo is fixed. I'm not sure where the indentation issues are, but if you point them out I will at least fix the ones in this PR.

File: scripts/exgfs_nceppost_cpl.sh [hunk 2]
```diff
@@ -155,21 +164,70 @@ then
 
      if test "$SENDDBN" = 'YES'
      then
-       #$DBNROOT/bin/dbn_alert MODEL GFS_MSC_sfcanl $job $COMOUT/${PREFIX}sfcanl
-       #$DBNROOT/bin/dbn_alert MODEL GFS_SA $job $COMIN/${PREFIX}atmanl
-       $DBNROOT/bin/dbn_alert MODEL GFS_MASTER $job $COMOUT/${MASTERANL}
+       $DBNROOT/bin/dbn_alert MODEL GFS_MSC_sfcanl $job $COMOUT/${PREFIX}sfcanl${SUFFIX}
+       $DBNROOT/bin/dbn_alert MODEL GFS_SA $job $COMOUT/${PREFIX}atmanl${SUFFIX}
+#alert removed in v15.0       $DBNROOT/bin/dbn_alert MODEL GFS_MASTER $job $COMOUT/${MASTERANL}
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_nceppost_cpl.sh:169]: Indentation (though it can probably just be deleted)

File: scripts/exgfs_nceppost_cpl.sh [hunk 3]
```diff
@@ -302,45 +359,57 @@ do
...
 	if [ $GRIBVERSION = 'grib2' ] ; then
-	  $DBNROOT/bin/dbn_alert MODEL GFS_MASTER $job $COMOUT/${MASTERFL}
-          if test "$PGBF" = 'YES'
-          then
-            $DBNROOT/bin/dbn_alert MODEL GFS_PGB2_0P25 $job $COMOUT/${PREFIX}pgrb2.0p25.f${fhr}
-            $DBNROOT/bin/dbn_alert MODEL GFS_PGB2_0P25_WIDX $job $COMOUT/${PREFIX}pgrb2.0p25.f${fhr}.idx
-            $DBNROOT/bin/dbn_alert MODEL GFS_PGB2B_0P25 $job $COMOUT/${PREFIX}pgrb2b.0p25.f${fhr}
-            $DBNROOT/bin/dbn_alert MODEL GFS_PGB2B_0P25_WIDX $job $COMOUT/${PREFIX}pgrb2b.0p25.f${fhr}.idx
-          fi
+            if [ $INLINE_POST = ".false." ]; then
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_nceppost_cpl.sh:368]: Indentation not consistent

File: scripts/exgfs_nceppost_cpl.sh [hunk 4]
```diff
@@ -302,45 +359,57 @@ do
...
-#           Only master grib files are needed for the hourly files
-#      if [ $x3 -eq 0 ] ; then
-       export fhr
-       $USHgfs/gfs_transfer.sh
-#      fi
+	
+	if test "$SENDDBN" = 'YES'
+	then
+	    if [ $GRIBVERSION = 'grib2' ] ; then
+#alert removed in v15.0		$DBNROOT/bin/dbn_alert MODEL GFS_MASTER $job $COMOUT/${MASTERFL}
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_nceppost_cpl.sh:377]: Indentation (remove?)

File: scripts/exgfs_nceppost_cpl.sh [hunk 5]
```diff
@@ -436,65 +499,56 @@ do
...
+          cp $PARMpost/gtg_imprintings.txt gtg_imprintings.txt
+
+          export PGBOUT=wafsfile
+          export PGIOUT=wafsifile
+
+          # WAFS data is processed:
+          #   hourly if fhr<=24
+          #   every 3 forecast hour if 24<fhr<=48
+          #   every 6 forecast hour if 48<fhr<=120
+	  if [  $fhr -le 24  ] ; then
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_nceppost_cpl.sh:527]: Indentation not consistent with previous

File: scripts/exgfs_nceppost_cpl.sh [hunk 6]
```diff
@@ -436,65 +499,56 @@ do
...
+               $POSTGPSH
+	  fi
 
-          SPECIALFL=${PREFIX}gtg.grb2
-          SPECIALFLIDX=${PREFIX}gtg.grb2i
+          export err=$?; err_chk
 
+          if [ -e $PGBOUT ]
+          then
           if test $SENDCOM = "YES"
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_nceppost_cpl.sh:541]: Needs to be indented from outer control block

File: ush/forecast_postdet.sh [hunk 7]
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
...
+        affix="nc"
+        if [ "$OUTPUT_FILE" = "nemsio" ]; then
+          affix="nemsio"
+        fi 
+
+	if [ $QUILTING = ".true." -a $OUTPUT_GRID = "gaussian_grid" ]; then
+	  fhr=$FHMIN
+	  while [ $fhr -le $FHMAX ]; do
+	    FH3=$(printf %03i $fhr)
+            FH2=$(printf %02i $fhr)
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:447]: Indentation doesn't match surrounding

File: ush/forecast_postdet.sh [hunk 8]
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
...
+	    atmi=atmf${FH3}.$affix
+	    sfci=sfcf${FH3}.$affix
+	    logi=logf${FH3}
+            pgbi=GFSPRS.GrbF${FH2}
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:451]: Indentation doesn't match surrounding

File: ush/forecast_postdet.sh [hunk 9]
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
...
+            flxi=GFSFLX.GrbF${FH2}
+	    atmo=$memdir/${CDUMP}.t${cyc}z.atmf${FH3}.$affix
+	    sfco=$memdir/${CDUMP}.t${cyc}z.sfcf${FH3}.$affix
+	    logo=$memdir/${CDUMP}.t${cyc}z.logf${FH3}.txt
+            pgbo=$memdir/${CDUMP}.t${cyc}z.master.grb2f${FH3}
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:456]: Indentation doesn't match surrounding

File: ush/forecast_postdet.sh [hunk 10]
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
...
+            flxo=$memdir/${CDUMP}.t${cyc}z.sfluxgrbf${FH3}.grib2
+	    eval $NLN $atmo $atmi
+	    eval $NLN $sfco $sfci
+	    eval $NLN $logo $logi
+            if [ $WRITE_DOPOST = ".true." ]; then
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:461]: Indentation doesn't match surrounding

File: ush/forecast_postdet.sh [hunk 11]
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
...
+	    if [ $FHMAX_HF -gt 0 -a $FHOUT_HF -gt 0 -a $fhr -lt $FHMAX_HF ]; then
+	      FHINC=$FHOUT_HF
+	    fi
+	    fhr=$((fhr+FHINC))
+	  done
+	else
+	  for n in $(seq 1 $ntiles); do
+	    eval $NLN nggps2d.tile${n}.nc       $memdir/nggps2d.tile${n}.nc
+	    eval $NLN nggps3d.tile${n}.nc       $memdir/nggps3d.tile${n}.nc
+            eval $NLN grid_spec.tile${n}.nc     $memdir/grid_spec.tile${n}.nc
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:475]: Indentation doesn't match surrounding

File: ush/inter_flux.sh [hunk 12]
```diff
@@ -45,9 +45,16 @@ else
 fi
 
 #---------------------------------------------------------------
-  $WGRIB2 $PGBOUT $option1 $option21 $option22 $option23 $option24 \
+
+  if [ $INLINE_POST = ".false." ]; then
+    $WGRIB2 $PGBOUT $option1 $option21 $option22 $option23 $option24 \
+                          $option25 $option26 $option27 $option28 \
```
Comment (WalterKolczynski-NOAA) [inline ush/inter_flux.sh:51]: Indentation for line continuations should be consistent with each other (L51 & 52)

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Indentation issues noted.

File: scripts/exgfs_nceppost_cpl.sh (hunk 5 above)
Comment (JessicaMeixner-NOAA) [reply scripts/exgfs_nceppost_cpl.sh:527]: This is copied from the NCEP post library -- it should just be linked in but we have to add one extra loop for the flux.1p00, therefore if the indentation should be fixed, it should be fixed in the NCEP Post library and copied in here.  This file shouldn't even exist in this repo, but it's the only work around we have for now.

File: ush/inter_flux.sh (hunk 12 above)
Comment (JessicaMeixner-NOAA) [reply ush/inter_flux.sh:51]: should be fixed now

File: ush/forecast_postdet.sh (hunk 7 above)
Comment (JessicaMeixner-NOAA) [reply ush/forecast_postdet.sh:447]: These were all issues of a mix of tabs and spaces being used, they line up when looking at the routines in vi but not when looking at it in GitHub.  All tabs were replaced by spaces in this routine.  All ush/forecast_postdet.sh indentation issues should be fixed.

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA with the exception of scripts/exgfs_nceppost_cpl.sh which should be linked to ../sorc/gfs_post.fd/scripts/exgdas_atmos_nceppost.sh and therefore carries any indentation issues over from that file, the indentation issues should now be fixed.

File: ush/forecast_postdet.sh (hunk 7 above)
Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh:447]: At some point, all spaces will need to be replaced with tabs in all g-w bash scripts, but that is not something you need to worry about. This is fine for now.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**